### PR TITLE
LPS-170565 Error in Object widget when sorting data by a Rich Text column

### PR DESCRIPTION
**Pull request by sontruongces**
Hi team,

This is the fix for the [LPS-170565](https://issues.liferay.com/browse/LPS-170565), the original has passed my review at sontruongces#69. Please help me to review this PR and give feedback.

Thanks
cc: @TaiPhamLR

### CODE REVIEW

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java [hunk 1]
```diff
@@ -120,7 +120,8 @@ protected Optional<EntityField> getEntityFieldOptional(
 				fieldName.substring(complexTypeName.length() + 1));
 		}
 
-		return Optional.ofNullable(entityFieldsMap.get(fieldName));
+		return Optional.ofNullable(
+			_getCorrectEntityField(entityFieldsMap, fieldName));
```
Comment (carolmariaabb) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java:124]: @TaiPhamLR @sontruongces can we inline here? Something like:

```
Optional.ofNullable(
	entityFieldsMap.get(ieldName)
).orElseGet(
	() -> {
		if (fieldName.endsWith("RawText")) {
			return entityFieldsMap.get(
					com.liferay.portal.kernel.util.StringUtil.replaceLast(
						fieldName, "RawText", StringPool.BLANK));
		}

		return null;
	}
);
```

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java [hunk 2]
```diff
@@ -183,12 +184,30 @@ protected boolean isAscending(String orderBy) {
...
+			return entityFieldsMap.get(fieldName);
+		}
+
+		if (fieldName.endsWith(_RAW_TEXT)) {
+			return entityFieldsMap.get(
+				com.liferay.portal.kernel.util.StringUtil.replaceLast(
+					fieldName, _RAW_TEXT, StringPool.BLANK));
+		}
+
+		return null;
```
Comment (carolmariaabb) [inline modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java:200]: @TaiPhamLR @sontruongces could you explain the reason why you added this code?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/petra/sql/dsl/expression/OrderByExpressionUtil.java [hunk 3]
```diff
@@ -47,8 +49,24 @@ public static OrderByExpression[] getOrderByExpressions(
...
+				catch (UnsupportedOperationException
+							unsupportedOperationException) {
+
+					Table<?> table = objectFieldLocalService.getTable(
+						objectDefinitionId, fieldName);
+
+					column = table.getColumn(fieldName + StringPool.UNDERLINE);
+
+					if (column == null) {
+						throw unsupportedOperationException;
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/petra/sql/dsl/expression/OrderByExpressionUtil.java:67]: @TaiPhamLR @sontruongces could you explain the reason why you added this code?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by samuelromeiro** [comment]
Hey @TaiPhamLR can you check Carol's comments and the conflicts in this PR. 
I'll close and for now. When you finish the changes, can you resend it, please?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/petra/sql/dsl/expression/OrderByExpressionUtil.java (hunk 3 above)
Comment (TaiPhamLR) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/petra/sql/dsl/expression/OrderByExpressionUtil.java:67]: @sontruongces had created PR wrong branch. sorry about that. He'll create new PR

**Pull request closed without merging**